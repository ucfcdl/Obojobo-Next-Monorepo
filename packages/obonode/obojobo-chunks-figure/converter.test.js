jest.mock('obojobo-document-engine/src/scripts/oboeditor/util/text-util')

import Converter from './converter'

describe('Figure Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return {}
				}
			},
			text: 'mockText',
			nodes: [
				{
					leaves: [
						{
							text: 'mockText',
							marks: [
								{
									type: 'b',
									data: {}
								}
							]
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with caption, size and triggers', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {
				size: 'large',
				triggers: 'mock-triggers',
				textGroup: [
					{
						text: { value: 'mockCaption' }
					}
				]
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
