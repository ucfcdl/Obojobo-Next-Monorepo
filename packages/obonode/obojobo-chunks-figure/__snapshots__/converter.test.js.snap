// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {},
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "leaves": Array [
        Object {
          "text": "",
        },
      ],
      "object": "text",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`Figure Converter oboToSlate converts an OboNode to a Slate node with caption, size and triggers 1`] = `
Object {
  "data": Object {
    "content": Object {
      "size": "large",
      "textGroup": Array [
        Object {
          "text": Object {
            "value": "mockCaption",
          },
        },
      ],
      "triggers": "mock-triggers",
    },
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "leaves": undefined,
      "object": "text",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`Figure Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "textGroup": Array [
      Object {
        "data": null,
        "text": Object {
          "styleList": Array [],
          "value": "mockText",
        },
      },
    ],
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
